index.js

```diff
@@ -2,71 +2,75 @@
 
 // Cache the matrix. Note that if you not pass a limit this implementation will use a dynamically calculate one.
 
-module.exports = function(__this, that, limit) {
+module.exports = function(firstString, secondString, maxSteps) {
 
-  var thisLength = __this.length,
-      thatLength = that.length,
-      matrix = [];
+  const firstLength = firstString.length;
+  const secondLength = secondString.length;
+  const matrix = [];
 
-  // If the limit is not defined it will be calculate from this and that args.
-  limit = (limit || ((thatLength > thisLength ? thatLength : thisLength)))+1;
+  // If maxSteps is not provided, limit will be calculated from firstString and secondString args.
+  const limit = (maxSteps || ((firstLength > secondLength ? firstLength : secondLength))) + 1;
 
-  for (var i = 0; i < limit; i++) {
+  if (Math.abs(firstLength - secondLength) > limit) {
+    return prepare(limit);
+  } else if (firstLength === 0) {
+    return prepare(secondLength);
+  } else if (secondLength === 0) {
+    return prepare(firstLength);
+  }
+
+  for (let i = 0; i < limit; i++) {
     matrix[i] = [i];
     matrix[i].length = limit;
   }
-  for (i = 0; i < limit; i++) {
-    matrix[0][i] = i;
-  }
 
-  if (Math.abs(thisLength - thatLength) > (limit || 100)){
```

Comment:
⚠️ There was an arbitrary (undocumented) hardcoded limit of 100. I have removed it in this PR. ⚠️